bundler/spec/bundler/rubygems_integration_spec.rb

```diff
@@ -41,13 +40,43 @@
     end
     let(:fetcher) { double("gem_remote_fetcher") }
 
-    it "successfully downloads gem with retries" do
-      expect(Bundler::Retry).to receive(:new).with("download gem from #{uri}/").
-        and_return(bundler_retry)
-      expect(bundler_retry).to receive(:attempts).and_yield
-      expect(fetcher).to receive(:cache_update_path)
+    context "when uri is public" do
+      let(:uri) { Gem::URI.parse("https://foo.bar") }
 
-      Bundler.rubygems.download_gem(spec, uri, cache_dir, fetcher)
+      it "successfully downloads gem with retries" do
+        expect(Bundler::Retry).to receive(:new).with("download gem from #{uri}/").
+          and_return(bundler_retry)
+        expect(bundler_retry).to receive(:attempts).and_yield
+        expect(fetcher).to receive(:cache_update_path)
+
+        Bundler.rubygems.download_gem(spec, uri, cache_dir, fetcher)
+      end
+    end
+
+    context "when uri contains user and password" do
+      let(:uri) { Gem::URI.parse("https://user:password@foo.bar") }
+
+      it "successfully downloads gem with retries with filtered log" do
+        expect(Bundler::Retry).to receive(:new).with("download gem from https://user:REDACTED@foo.bar/").
+          and_return(bundler_retry)
+        expect(bundler_retry).to receive(:attempts).and_yield
+        expect(fetcher).to receive(:cache_update_path)
+
+        Bundler.rubygems.download_gem(spec, uri, cache_dir, fetcher)
+      end
+    end
+
+    context "when uri contains user" do
+      let(:uri) { Gem::URI.parse("https://user@foo.bar") }
```

Comment:
Out of context I would find this description confusing, I could easily imagine people going "huh, what's wrong about the user?"

I would consider rewording the description for keeping the context around, for being more obvious about it. What do you think of

```ruby 
    context "when uri contains userinfo part" do
      context "with user and password" 
      context "with token [as user]"
```

or just for this context,

```suggestion
    context "when uri has userinfo which is a token" do
      let(:uri) { Gem::URI.parse("https://token@foo.bar") }
```

(I think the former is slightly more informative)

Reply:
Great suggestion, you're absolutely right. I will add this 👍.